packages/pointer_interceptor/pointer_interceptor_web/example/integration_test/widget_test.dart

```diff
@@ -134,20 +112,29 @@ void main() {
     // simply allows the hit test to land on the platform view by making itself
     // hit test transparent.
     testWidgets('on background directly', (WidgetTester tester) async {
-      app.main();
-      await tester.pumpAndSettle();
+      await _fullyRenderApp(tester);
 
-      final html.Element element =
+      final web.Element element =
           _getHtmlElementAt(tester.getTopLeft(backgroundFinder));
 
       expect(element.id, 'background-html-view');
     });
   });
 }
 
+Future<void> _fullyRenderApp(WidgetTester tester) async {
+  app.main();
+  // Because the app widget is not injected through the `tester`, we need to
+  // await a bunch of frames (~5) so the app is fully rendered, and the
+  // platform views injected in the DOM.
+  for (int i = 0; i < 5; i++) {
+    await tester.pump(const Duration(milliseconds: 16));
+  }
```

Comment:
Is there a less flaky way to wait for render? (I'm assuming no)

Reply:
There's a `tester.pumpFrames(Widget, Duration time, [Duration interval])` but the internal implementation is just a for loop that does `tester.pump` `time/interval` times.

HOWEVER, now that I'm not using custom factories to render, I might be able to just `pumpWidget` with the app, and just pump a couple of frames directly! I'll give it a shot, thanks @tarrinneal!

Reply:
(This seems to work. I've fixed the example app so we can pump its main Widget from the tester, instead of calling its "main", and now this just needs a couple of `pump` calls to get the platform view rendered into the DOM).